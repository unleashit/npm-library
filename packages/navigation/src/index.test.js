import * as React from 'react';
import { mount } from 'enzyme';
import NavigationContainer from '.';

describe('<Navigation />', () => {
  let wrapper;
  const props = {
    links: [
      {
        text: 'Home',
        url: '/home',
        // icon: 'https://img.icons8.com/material/420/home-page.png',
        // iconPosition: 'left',
      },
      {
        text: 'Products',
        url: '/products',
      },
      {
        text: 'Services',
        url: '/services',
      },
      {
        text: 'About',
        url: '/about',
        attrs: {
          target: '_blank',
          rel: 'noopener noreferrer',
        },
      },
    ]
  };

  beforeEach(() => {
    wrapper = mount(<NavigationContainer {...props} />);
  });

  it.only('renders without crashing', () => {
    expect(wrapper.find('.unl-navigation__container')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
