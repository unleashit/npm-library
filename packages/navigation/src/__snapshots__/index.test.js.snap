// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> renders without crashing 1`] = `
<Navigation
  direction="horizontal"
  links={
    Array [
      Object {
        "text": "Home",
        "url": "/home",
      },
      Object {
        "text": "Products",
        "url": "/products",
      },
      Object {
        "text": "Services",
        "url": "/services",
      },
      Object {
        "attrs": Object {
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "text": "About",
        "url": "/about",
      },
    ]
  }
  template="clean"
>
  <nav
    className="container unl-navigation__container clean unl-navigation__container--clean"
  >
    <NavLinks
      cssModuleStyle={null}
      links={
        Array [
          Object {
            "text": "Home",
            "url": "/home",
          },
          Object {
            "text": "Products",
            "url": "/products",
          },
          Object {
            "text": "Services",
            "url": "/services",
          },
          Object {
            "attrs": Object {
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "text": "About",
            "url": "/about",
          },
        ]
      }
    >
      <ul
        className="navList unl-navigation__nav-list"
      >
        <NavLink
          cssModuleStyle={null}
          key="Home"
          text="Home"
          url="/home"
        >
          <li
            className="linkItem  unl-navigation__link-item "
          >
            <a
              className="link unl-navigation__link"
              href="/home"
            >
              Home
            </a>
          </li>
        </NavLink>
        <NavLink
          cssModuleStyle={null}
          key="Products"
          text="Products"
          url="/products"
        >
          <li
            className="linkItem  unl-navigation__link-item "
          >
            <a
              className="link unl-navigation__link"
              href="/products"
            >
              Products
            </a>
          </li>
        </NavLink>
        <NavLink
          cssModuleStyle={null}
          key="Services"
          text="Services"
          url="/services"
        >
          <li
            className="linkItem  unl-navigation__link-item "
          >
            <a
              className="link unl-navigation__link"
              href="/services"
            >
              Services
            </a>
          </li>
        </NavLink>
        <NavLink
          attrs={
            Object {
              "rel": "noopener noreferrer",
              "target": "_blank",
            }
          }
          cssModuleStyle={null}
          key="About"
          text="About"
          url="/about"
        >
          <li
            className="linkItem  unl-navigation__link-item "
          >
            <a
              className="link unl-navigation__link"
              href="/about"
              rel="noopener noreferrer"
              target="_blank"
            >
              About
            </a>
          </li>
        </NavLink>
      </ul>
    </NavLinks>
  </nav>
</Navigation>
`;
